server/test/durablejobs/jobs/MigratePrimaryApplicantInfoJobTest.java

```diff
@@ -0,0 +1,124 @@
+    ImmutableMap<String, String> settings =
+        settingsService.loadSettings().toCompletableFuture().join().get();
+    ImmutableMap.Builder<String, String> newSettings = ImmutableMap.builder();
+    for (var entry : settings.entrySet()) {
+      if (!entry.getKey().equals("PRIMARY_APPLICANT_INFO_QUESTIONS_ENABLED")) {
+        newSettings.put(entry);
+      }
+    }
+    newSettings.put("PRIMARY_APPLICANT_INFO_QUESTIONS_ENABLED", setFlag.toString());
+    settingsService.updateSettings(newSettings.build(), "test");
+    transaction.commit();
+    transaction.close();
+    transaction = database.beginTransaction(TxIsolation.SERIALIZABLE);
+
+    PersistedDurableJobModel job =
+        new PersistedDurableJobModel(
+            DurableJobName.MIGRATE_PRIMARY_APPLICANT_INFO.toString(), Instant.ofEpochMilli(0));
+    MigratePrimaryApplicantInfoJob migrateJob =
+        new MigratePrimaryApplicantInfoJob(
+            job, instanceOf(AccountRepository.class), settingsService, instanceOf(Config.class));
+    migrateJob.run();
+    transaction.commit();
+    transaction.close();
+  }
+
+  @Test
+  public void run_doesNotMigrateDataWhenFlagIsOn() {
+    ApplicantModel applicant = createApplicantWithWellKnownPathData(/* withPaiData= */ true);
+    runJob(/* setFlag= */ true);
+    applicant.refresh();
+    assertThat(applicant.getFirstName().get()).isEqualTo("Kathryn");
+    assertThat(applicant.getMiddleName()).isEmpty();
+    assertThat(applicant.getLastName().get()).isEqualTo("Janeway");
+    assertThat(applicant.getDateOfBirth().get()).isEqualTo("2328-05-20");
+    assertThat(applicant.getPhoneNumber().get()).isEqualTo("2066842489");
+  }
+
+  @Test
+  public void run_migratesWhenPaiDataIsEmpty() {
+    ApplicantModel applicant = createApplicantWithWellKnownPathData(/* withPaiData= */ false);
+    runJob();
+    applicant.refresh();
+    assertThat(applicant.getFirstName().get()).isEqualTo("Jean");
+    assertThat(applicant.getMiddleName().get()).isEqualTo("Luc");
+    assertThat(applicant.getLastName().get()).isEqualTo("Picard");
+    assertThat(applicant.getDateOfBirth().get()).isEqualTo("2305-07-13");
+    assertThat(applicant.getPhoneNumber().get()).isEqualTo("5038234000");
+    assertThat(applicant.getEmailAddress().get()).isEqualTo("account@email.com");
+  }
+
+  @Test
+  public void run_migratesWhenPaiDataIsAlreadyPopulated() {
+    ApplicantModel applicant = createApplicantWithWellKnownPathData(/* withPaiData= */ true);
+    runJob();
+    applicant.refresh();
+    assertThat(applicant.getFirstName().get()).isEqualTo("Jean");
+    assertThat(applicant.getMiddleName().get()).isEqualTo("Luc");
+    assertThat(applicant.getLastName().get()).isEqualTo("Picard");
+    assertThat(applicant.getDateOfBirth().get()).isEqualTo("2305-07-13");
+    assertThat(applicant.getPhoneNumber().get()).isEqualTo("5038234000");
+    assertThat(applicant.getEmailAddress().get()).isEqualTo("applicant@email.com");
+  }
+}
```

Comment:
question (blocking): Is it worth adding additional tests for:
 - When only some of the WKPs are filled in, those existing WKPs are migrated and we don't try to migrate WKPs that aren't filled in
 - Email is only migrated if it's not already filled in

Reply:
Thanks for the new tests! My only nitpick would be putting the email test in a separate test case - that way, even if the `someWpkData` test fails and stops early, the email test is still run. Totally fine to keep it as-is though.